openssl/src/cms.rs

```diff
@@ -202,11 +281,161 @@ impl CmsContentInfo {
     }
 }
 
+#[cfg(ossl110)]
+mod recipient_info {
+    use asn1::Asn1IntegerRef;
+    use stack::Stackable;
+    use x509::X509NameRef;
+    use foreign_types::Opaque;
+    use super::*;
+
+    impl CmsContentInfoRef {
+        /// Given that we're dealing with the `EnvelopedData`, returns the recipients of the message.
+        ///
+        /// OpenSSL documentation at [`CMS_get0_RecipientInfos`]
+        ///
+        /// [`CMS_get0_RecipientInfos`]: https://www.openssl.org/docs/manmaster/man3/CMS_RecipientInfo_decrypt.html
+        pub fn get_recipient_infos(&self) -> Result<&StackRef<CmsRecipientInfo>, ErrorStack> {
+            unsafe {
+                let recipient_infos = cvt_p(ffi::CMS_get0_RecipientInfos(self.as_ptr()))?;
+                Ok(StackRef::from_ptr(recipient_infos))
+            }
+        }
+    }
+
+    /// An owned version of [`CmsRecipientInfoRef`]
+    ///
+    /// [`CmsRecipientInfoRef`]:struct.CmsRecipientInfoRef.html
+    pub struct CmsRecipientInfo(*mut ffi::CMS_RecipientInfo);
```

Comment:
These can be defined by the macro like we do for other types.

Reply:
I assume we're talking about `foreign_type_and_impl_send_sync` macro, right? I considered its usage, however the problem is that it requires us to specify a drop function (https://github.com/sfackler/rust-openssl/blob/master/openssl/src/macros.rs#L137), and there is no dedicated drop function for `CmsRecipientInfo` as far as I know.